openfecli/commands/gather.py

```diff
@@ -131,10 +131,11 @@ def _generate_bad_legs_error_message(set_vals, ligpair):
 def _parse_raw_units(results: dict) -> list[tuple]:
     # grab individual unit results from master results dict
     # returns list of (estimate, uncertainty) tuples
-    list_of_pur = list(results['protocol_result']['data'].values())[0]
+    list_of_pur = list(results['protocol_result']['data'].values())
 
-    return [(pu['outputs']['unit_estimate'],
-             pu['outputs']['unit_estimate_error'])
+    # could add to each tuple pu[0]["source_key"] for repeat ID
+    return [(pu[0]['outputs']['unit_estimate'],
```

Comment:
Could you maybe provide some context for this change?

As far as I remember, the zero list index on line 134 is to deal with avoiding to do it here twice - that being said I could be convinced that this is a better approach with the right context.

Reply:
`list(results['protocol_result']['data'].values())` in line 134 contains an item for each ProtocolUnit, so by taking the zero list index there only the first PU is further processed. Each item in `list(results['protocol_result']['data'].values())` is a list itself, with length 1 (afaik), so that's why pu[0] must be done to access the single dictionary within.

![image](https://github.com/OpenFreeEnergy/openfe/assets/74412019/b654131b-3d08-4776-b9dc-f75e2ae1840a)